tools/cabana/chartswidget.cc

```diff
@@ -166,8 +165,11 @@ void ChartsWidget::setMaxChartRange(int value) {
 }
 
 void ChartsWidget::updateToolBar() {
+  if (docking) setColumnCount(0);
+  columns_lb_action->setVisible(!docking);
+  columns_cb_action->setVisible(!docking);
```

Comment:
how about call updateToolbar in resizeEvent, and set them to visible if rect.width() > CHART_MIN_WIDTH * 2?

Reply:
That's a good solution! I was worried about the dropdown disappearing but still showing multiple columns.